account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,1130 @@
+            )
+            logger.debug("Starting to import the following XML file:")
+            logger.debug(pretty_xml_string)
+            parsed_inv = self.parse_xml_invoice(xml_root)
+            if parsed_inv is False:
+                raise UserError(
+                    _(
+                        "This type of XML invoice is not supported. "
+                        "Did you install the module to support this type "
+                        "of file?"
+                    )
+                )
+        # Fallback on PDF
+        else:
+            parsed_inv = self.parse_pdf_invoice(file_data)
+        if "attachments" not in parsed_inv:
+            parsed_inv["attachments"] = {}
+        parsed_inv["attachments"][invoice_filename] = invoice_file_b64
+        # pre_process_parsed_inv() will be called again a second time,
+        # but it's OK
+        pp_parsed_inv = self.pre_process_parsed_inv(parsed_inv)
+        return pp_parsed_inv
+
+    @api.model
+    def pre_process_parsed_inv(self, parsed_inv):
+        if parsed_inv.get("pre-processed"):
+            return parsed_inv
+        parsed_inv["pre-processed"] = True
+        if "chatter_msg" not in parsed_inv:
+            parsed_inv["chatter_msg"] = []
+        if parsed_inv.get("type") in ("out_invoice", "out_refund"):
+            return parsed_inv
+        prec_ac = self.env["decimal.precision"].precision_get("Account")
+        prec_pp = self.env["decimal.precision"].precision_get("Product Price")
+        prec_uom = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        if "amount_tax" in parsed_inv and "amount_untaxed" not in parsed_inv:
+            parsed_inv["amount_untaxed"] = (
+                parsed_inv["amount_total"] - parsed_inv["amount_tax"]
+            )
+        elif "amount_untaxed" not in parsed_inv and "amount_tax" not in parsed_inv:
+            # For invoices that never have taxes
+            parsed_inv["amount_untaxed"] = parsed_inv["amount_total"]
+        # Support the 2 refund methods; if method a) is used, we convert to
+        # method b)
+        if not parsed_inv.get("type"):
+            parsed_inv["type"] = "in_invoice"  # default value
+        if (
+            parsed_inv["type"] == "in_invoice"
+            and float_compare(parsed_inv["amount_total"], 0, precision_digits=prec_ac)
+            < 0
+        ):
+            parsed_inv["type"] = "in_refund"
+            for entry in ["amount_untaxed", "amount_total"]:
+                parsed_inv[entry] *= -1
+            for line in parsed_inv.get("lines", []):
+                line["qty"] *= -1
+                if "price_subtotal" in line:
+                    line["price_subtotal"] *= -1
+        # Handle taxes:
+        self._pre_process_parsed_inv_taxes(parsed_inv)
+        # Handle rounding:
+        for line in parsed_inv.get("lines", []):
+            line["qty"] = float_round(line["qty"], precision_digits=prec_uom)
```

Comment:
There is an exception possibility here:

Invoice line qty = 0,5 / price = 100
Product Unit of Measure = 0 

When importing the quantity will be set to 1, but the price is not changed accordingly. Result is an unclear debit/credit error from Odoo.